main.js

```diff
@@ -1,10 +1,297 @@
-const hamberger = document.getElementById('hamberger');
-const menu = document.getElementById('menu');
+const menu = document.getElementById('nav-elements');
+const menuOpenClose = document.getElementById('menu');
+const logo = document.getElementById('logo');
+const mobileNavList = document.querySelectorAll('.mobile-nav-list');
 
-const mobileMenu = () => {
+function togglerMenu() {
   menu.classList.toggle('active');
-  hamberger.classList.toggle('close');
-};
+  menuOpenClose.classList.toggle('open');
+  menuOpenClose.classList.toggle('close');
+  logo.classList.toggle('hide');
+  document.body.classList.toggle('mobile-nav-open');
+}
 
-hamberger.addEventListener('click', mobileMenu);
-menu.addEventListener('click', mobileMenu);
\ No newline at end of file
+mobileNavList.forEach((element) => {
+  element.addEventListener('click', togglerMenu);
+});
+
+// form validation
+const form = document.querySelector('.contact-form');
+const emailInput = form.querySelector('#email');
+const errorMessage = form.querySelector('#error-message');
+
+form.addEventListener('submit', (event) => {
+  if (emailInput.value !== emailInput.value.toLowerCase()) {
+    event.preventDefault();
+    errorMessage.textContent = 'Please enter a lower case email address.';
+  }
+});
+
+form.addEventListener('input', () => {
+  const formData = Object.fromEntries(new FormData(form).entries());
+  localStorage.setItem('UserData', JSON.stringify(formData));
+});
+
+const savedFormData = JSON.parse(localStorage.getItem('UserData'));
+if (savedFormData) {
+  Object.entries(savedFormData).forEach(([name, value]) => {
+    const input = form.elements[name];
+    if (input) {
+      input.value = value;
+    }
+  });
+}
+
+//  Work Projects Data
+
+const projects = [
+  {
+    main: true,
+    name: 'Multi-Post Stories',
+    image: './images/img-placeholder-desktop.svg',
+    image2: './images/ImgPlaceholder.png',
+    image3: './images/Snapshoot Portfolio.png',
+    description:
+      'A daily selection of privately personalized reads; no accounts or sign-ups required. This has been the industrys standard dummy text ever since the 1500s, when an unknown printer took a standard dummy text.',
+    tags: ['HTML', 'CSS', 'Bootstrap', 'Ruby'],
+    demoLink: '',
+    codeLink: '',
+  },
+
+  {
+    main: false,
+    name: 'Profesional Art Printing Data More',
+    image: './images/Last-card.svg',
+    image2: './images/ImgPlaceholder.png',
+    image3: './images/Snapshoot Portfolio.png',
+    description:
+      'A daily selection of privately personalized reads; no accounts or sign-ups required. Has been the industrys standard.',
+    tags: ['HTML', 'Bootstrap', 'Ruby'],
+    demoLink: '',
+    codeLink: '',
+  },
+  {
+    main: false,
+    name: 'Data Dashboard Healthcare',
+    image: './images/Last-card.svg',
```

Comment:
[REQUIRED] I noticed that when hovering over any of the project cards the card disappears and I see only the `see project` button. This opposes the Figma design template.

![image](https://github.com/ClarenceAtim/Portfolio/assets/62938735/86341ec1-f169-4049-bca8-5b0635200b33)

[REQUIRED] Kindly ensure that there is a detail pop-up when the `see project` button is clicked. Also, ensure that this pop-up has a close button and ensure that it is responsive. These are requirements for this milestone.